### ISSUE TITLE
Infinite hanging on reply

### ISSUE THREAD

**Issue by Dementiy** [pgiri/dispy#168]
Hello.

From time to time `dispynode` hanging on sending results back to cluster. For example:

```
dispynode - New computation "0343b6d3e0e6e9f36a32151bbdc35fddd6ff6d97" from 127.0.0.1
dispynode - New job id 264044360 from 10.57.46.38/127.0.0.1
```

Hanging (I'm sure that job is done).... But if I press **Enter** than results are sending back to cluster:

```
dispynode - Sending result for job 264044360 (11)
dispynode - New job id 263757896 from 10.57.46.38/127.0.0.1
```

I can't figure out what the reason of this behaviour.

My compute function is:

```python
def compute(args):
    import subprocess
    p = subprocess.Popen(args, stdout=subprocess.DEVNULL)
    p.wait()
    return p.returncode
```

System Info:
```
Windows 10
Python 3.6
dispy==4.10.3
```

**Comment by pgiri** [comment]
See note around line 2852 in dispynode.py. Is it possible for you to upgrade Python to latest 3.6 or even 3.7?

**Comment by pgiri** [comment]
Well, you mentioned that if you hit `Enter`, results are being sent to the cluster, which is what I observed when I was trying to deal with issues with Windows (quote some time ago). I wrote some test cases that showed that issue is with Python 3.6 and modules, even without using dispy. As noted, this seems to have been partially fixed in later version of 3.6. I haven't tested with 3.7. If you have separate test environment, you can try latest versions of Python that you can.

**Comment by Dementiy** [comment]
Perhaps the problem with hanging was caused by [QuickEdit](https://stackoverflow.com/questions/30418886/how-and-why-does-quickedit-mode-in-command-prompt-freeze-applications) mode in PowerShell or maybe it's a Windows [bug](https://bugs.python.org/issue26744). Now I'm using [Git Bash](https://gitforwindows.org/) and it works better.
